api/src/modules/countries/countries.service.ts

```diff
@@ -1,47 +1,45 @@
-import { Injectable } from '@nestjs/common';
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
 import { Country } from '@shared/dto/countries/country.entity';
 import { CreateCountryDto } from '@shared/dto/countries/create-country.dto';
 import { Repository } from 'typeorm';
 
 @Injectable()
 export class CountriesService {
-  constructor(private countryRepository: Repository<Country>) {}
-  private readonly country: Country[] = [];
+  constructor(@InjectRepository(Country)  private countryRepository: Repository<Country>) {}
 
-  create(country: CreateCountryDto) {
-    this.countryRepository.save(country);
+  async save(createCountryDto: CreateCountryDto) {
+    return this.countryRepository.save(createCountryDto);
   }
 
-  findAll(): Country[] {
-    return this.countryRepository.findAll();
+  async find(): Promise<Country[]> {
```

Comment:
`Promise<Country[]>` is not needed. It's implicit in `return this.countryRepository.find();`